pkg/operator/targetconfigcontroller/targetconfigcontroller.go

```diff
@@ -114,8 +116,9 @@ func createTargetConfig(ctx context.Context, c TargetConfigController, recorder
 	if err != nil {
 		return false, err
 	}
-	if err := checkExternalDependencies(ctx, c.configMapLister, recorder); err != nil {
-		errors = append(errors, err)
+	if err := checkExternalDependencies(ctx, c.configMapLister, c.infrastructureLister, recorder); err != nil {
+		recorder.Warning("MissingRequiredResource", err.Error())
```

Comment:
Will `checkExternalDependencies` ever return an error that does not indicate a missing resource?

Reply:
yeah, not in love with that.

Reply:
updated.